modules/siddhi-core/src/main/java/io/siddhi/core/table/Table.java

```diff
@@ -127,6 +127,10 @@ protected abstract void init(TableDefinition tableDefinition, StreamEventFactory
                                  StreamEventCloner storeEventCloner, ConfigReader configReader,
                                  SiddhiAppContext siddhiAppContext, RecordTableHandler recordTableHandler);
 
+    public SiddhiAppContext getSiddhiAppContext() {
```

Comment:
Why we need to get SiddhiAppContext from a table don't the relevant object has access to that during its initialization?